### PR TITLE
Add `unban` command

### PR DESCRIPTION
**Pull request by marqdevx**

### CODE REVIEW

File: src/adminsystem.cpp [hunk 1]
```diff
@@ -159,6 +159,55 @@ CON_COMMAND_CHAT_FLAGS(ban, "ban a player", ADMFLAG_BAN)
...
+	int iNumClients = 0;
+	int pSlot[MAXPLAYERS];
+
+	ETargetType nType = g_playerManager->TargetPlayerString(iCommandPlayer, args[1], iNumClients, pSlot);
+
+	if (!iNumClients)
+	{
+		ClientPrint(player, HUD_PRINTTALK, CHAT_PREFIX "Target not found.");
+		return;
+	}
```
Comment (xen-000) [inline src/adminsystem.cpp:180]: Banned players aren't on the server so you can't unban like that

Comment (marqdevx) [reply src/adminsystem.cpp:180]: True, I'll change it

**Comment by marqdevx** [comment]
Now it works with unconnected players, using target's `SteamID64`

**Comment by marqdevx** [comment]
I've just noticed that I think I might add some more code to be compliant with the previous `CAdminSystem::FindAndRemoveInfraction` which was doing `m_vecInfractions[i]->UndoInfraction(player);`

**Comment by mihaigsm2003** [comment]
after unban need reload infractions?

**Comment by marqdevx** [comment]
It's automated so the unbanned player is able to join right after.

File: src/adminsystem.cpp [hunk 2]
```diff
@@ -159,6 +159,26 @@ CON_COMMAND_CHAT_FLAGS(ban, "ban a player", ADMFLAG_BAN)
...
+
+	int iCommandPlayer = player ? player->GetPlayerSlot() : -1;
+	uint64 targetSteamID =  V_StringToUint64(args[1], -1);;
+
+	if(g_pAdminSystem->FindAndRemoveInfraction(targetSteamID, CInfractionBase::Ban)){
+		ClientPrint(player, HUD_PRINTTALK, CHAT_PREFIX "SeeamID %llu unbanned.", targetSteamID);
+	}else{
+		ClientPrint(player, HUD_PRINTTALK, CHAT_PREFIX "SeeamID %llu is not banned.", targetSteamID);
+	}
```
Comment (xen-000) [inline src/adminsystem.cpp:177]: Spelling. Also the braces should be on new lines, or you can remove them in this case.

**Pull request closed without merging**